addons/sale/report/sale_report.py

```diff
@@ -236,3 +240,12 @@ def _query(self):
     @property
     def _table_query(self):
         return self._query()
+
+    def action_open_order(self):
+        self.ensure_one()
+        return {
+            'type': 'ir.actions.act_window',
+            'res_model': 'sale.order',
+            'res_id': self.order_id.id,
+            'view_mode': 'form',
+        }
```

Comment:
I think this will not work if order if related to pos and give error :worried:

Reply:
Yes, I have made some changes in the action method now it will work if order is related to pos :100: .